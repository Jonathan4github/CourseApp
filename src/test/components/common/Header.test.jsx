/* eslint-disable */
import HeaderContainer, { Header } from '../../../components/common/Header';


const initialState = {
  courses: { allCourses: [{}, {}] },
  authors: { allAuthors: [{}, {}] }
};

const setup = () => {
  const store = storeMock(initialState);

  return {
    container: shallow(<HeaderContainer store={store} />),
    pressentation: shallow(<Header store={store} />)
  };
};

let shallowWrapper;

describe('<Header /> component test', () => {
  it('renders component without crashing', () => {
    shallowWrapper = setup().container;
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should has one div element', () => {
    shallowWrapper = setup().pressentation;

    expect(shallowWrapper.find('.header-container')).toBeDefined();
    expect(shallowWrapper.find('.header-container').children()).toHaveLength(2);
  });

  it('should show previously rolled value', () => {
    shallowWrapper = setup().container;
    const coursesLength = initialState.courses.allCourses.length;
    const authorsLength = initialState.authors.allAuthors.length;

    // test that the state values were correctly passed as props
    expect(shallowWrapper.props().numberOfCourses).toEqual(coursesLength);
    expect(shallowWrapper.props().numberOfAuthors).toEqual(authorsLength);
  });
});
