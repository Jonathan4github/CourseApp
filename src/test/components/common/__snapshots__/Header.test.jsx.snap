// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component test renders component without crashing 1`] = `
<Header
  dispatch={[Function]}
  numberOfAuthors={2}
  numberOfCourses={2}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
